### DISCUSSION TITLE
Set up relations with additional information

### DISCUSSION THREAD

**Discussion by hagealex** [Laravel-Backpack/community-forum#431] [Q&A (Help)] [answered]
Hello everyone,

currently I'm working on a hobby game project. I use Backpack as my "engine" where I set up all necessary data for the game. Like: NPCs, Items, Quests, Skills and so on.

Sometimes I have the case that something has a "requirement". For example: To learn "Skill B" you have to have "Skill A" at least at level 5. Or: To be able to use "Item A" you need to train "Skill C" to at least level 3.

With Backpack it is no problem to set up relations between all the models. But is it also possible somehow to add additional information to a relation?

So when I select the related item I would be able to add additional information to this specific relation.

As far as I know it is possible with Laravel to add additional columns to a pivot table for a hasMany || belongsToMany table.

I'm curious if this can be handled out of the box or if I have to make some custom logic :)

Kind regards

**Comment by karandatwani92** [comment, marked as answer]
Hi @Alexander-Hagemeier 

This is achievable using relationship field type, which is a part of PRO Package. Here is the docs [link](https://backpackforlaravel.com/docs/5.x/crud-fields#save-additional-data-to-pivot-table) pointing to the section which would help you save additional info.

You can also check the similar use case in demo. Please follow the this demo [link](https://demo.backpackforlaravel.com/admin/monster/140/edit#relationship) and go to relationship tab.

Here you will find the field highlighted in following image which does the same:
![Screenshot 2023-03-09 at 1 25 41 PM](https://user-images.githubusercontent.com/8214221/223957130-47f349e0-40bc-4fee-b262-452ecc3b081f.png)

You can check the code of working example in [Demo Repo](https://github.com/Laravel-Backpack/demo) & the Direct [link](https://github.com/Laravel-Backpack/demo/blob/main/app/Http/Controllers/Admin/MonsterCrudController.php#L722) to the field definition.

**Reply by hagealex** [reply]
Thank you @karandatwani92 I totally missed that while reading the documentation (#shame!).

Thanks so much for clarifying!
